src/reload/reload-client.js

```diff
@@ -36,58 +42,110 @@
   adoptedStyleSheetsCollector.add(adoptedStyleSheetsCollector.sheets);
   adoptedStyleSheetsCollector.sheets = [];
 
-  connect();
+  if (!canUseLeaderElection) {
+    connect();
+  } else {
+    channel = new BroadcastChannel('dvlp/reload');
+    channel.onmessage = (event) => {
+      if (event.data.type === 'reload') {
+        onReload();
+      } else if (event.data.type === 'refresh') {
+        onRefresh(event.data.event);
+      }
+    };
+
+    requestLeadership((success) => {
+      if (success) {
+        isLeader = true;
+        connect();
+      }
+    });
+  }
+
+  /**
+   * @param {(success: boolean) => void} callback
+   */
+  function requestLeadership(callback) {
+    navigator.locks.request(
+      'dvlp/reload',
+      { ifAvailable: false, mode: 'exclusive' },
+      (lock) => {
+        callback(lock !== null);
+        // Never relinquish until process exit
```

Comment:
Process exit sounds like closing the browser window, and not (afaik) the more likely case a location.reload